tools/toolchains/__init__.py

```diff
@@ -867,6 +870,26 @@ def mem_stats(self, map):
         map_csv = splitext(map)[0] + "_map.csv"
         memap.generate_output('csv-ci', map_csv)
 
+    # "Prefix headers" are automatically included by the compiler at the beginning of
+    # each source file.
+    # header_content - the content of the prefix header file.
+    def set_prefix_header_content(self, header_content):
+        self.prefix_header_content = header_content
+
+    # Return the location of the prefix header. This function will create the prefix
+    # header first if needed. The header will be written in a file called "mbed_prefix.h"
+    # located in the project's build directory.
+    # If prefix headers are not used (self.prefix_header_content is None), the function
+    # returns None
+    def get_prefix_header(self):
+        if self.prefix_header_content is None:
+            return None
+        prefix_file = join(self.build_dir, "mbed_prefix.h")
```

Comment:
As far as the toolchain is concerned, this is a prefix header, it doesn't know anything about the config system (which is completely external). However, that can change. I can modify some function names to make it clear that this feature is to be used by the config system, then modify the header name to `mbed_conf.h` (not `mbed_config.h`, since I believe that's a different mechanism that we introduced a while ago).